### DISCUSSION TITLE
How to handle stmt.invbal.availcash duplicating stmt.invposlist?

### DISCUSSION THREAD

**Discussion by emin63** [csingley/ofxtools#155] [Q&A]
I have some brokers who report a value in `stmt.invbal.availcash` and then provide positions in `stmt.invposlist` which duplicate `availcash`. For example, if I have some money market funds and bond funds, the total value is provided in `availcash`.

Most brokers do the more natural thing of reporting cash in `availcash` and then not reporting anything in `stmt.invposlist` which is already reported in `availcash`.

Does anyone have a suggestion on how to compute the total market value of positions and not double count? What I am currently doing is checking if `availcash` equals the some of market values in `stmt.invposlist` but that seems ugly.

Thanks.

**Comment by csingley** [comment]
OFX is terribly underspecified, especially for securities use.  Different brokers do different things with the data & don't adhere to conventions.  It's pretty much a necessity to create broker-specific handlers that call different data processors.  You're not likely to be able to get away with running all the different brokerage data down the same code path.
